### ISSUE TITLE
Characters able to counter while asleep, stunned, petrified or terrified

### ISSUE THREAD

**Issue by isotor** [DarkstarProject/darkstar#5848]
**_I have:_**

- [x] searched existing issues (http://github.com/darkstarproject/darkstar/issues/) to see if the issue I am posting has already been addressed or opened by another contributor
- [x] checked the commit log to see if my issue has been resolved since my server was last updated

**_Client Version_** (type `/ver` in game) **:** 30190328_2

**_Source Branch_** (master/stable) **:** master

**_Additional Information_** (Steps to reproduce/Expected behavior) **:** 

Characters (and mobs) should be unable to block, parry, or counter while under any of these four status effects.

**Comment by zach2good** [comment]
Possible relevant area:
https://github.com/DarkstarProject/darkstar/blob/master/src/map/entities/battleentity.cpp#L1563

**Comment by maxtherabbit** [comment]
do you have a source to verify how retail handles this?

**Comment by TeoTwawki** [comment]
I'm like 80% sure I have parried in my sleep on retail. The rest, no idea. Terror I think should stop everything. But we need some verification on retail both both. https://discord.gg/AVQRwv3

**Comment by isotor** [comment]
If this helps, check the section on Counterstance: https://www.ffxiah.com/forum/topic/36705/iipunch-monk-guide/#traits

> Keep in mind that you cannot counter while asleep, petrified, terrorized, or stunned. You also cannot counter mob area attacks, such as those from Ironclads and Mantis mobs, nor can you counter TP moves.

Thanks Nyu!

**Comment by isotor** [comment]
Not in this issue list but [this video](https://youtu.be/s6orP8M_nG8?t=72) confirms you can still proc guard while under any of the above effects *(not sure about terror but reasonably safe bet)*. Again props to Nyu for flagging.

**Comment by TeoTwawki** [comment]
ugh.. genomeffxi

..

/pukes

**Comment by maxtherabbit** [comment]
my assumption would be that if guard works, then all the other strictly defensive abilities will as well - which is how we have it currently

**Comment by isotor** [comment]
If you say, I provided a text guide from a fansite explicitly stating counter at the very least can't proc under these effects - it's going to be very difficult to arrange video proof of an effect *not* occurring under certain conditions.

**Comment by TeoTwawki** [comment]
1 important distinction here: guard parry etc have a skill, counter does not. this may be the thing that SE used to decide which things happen at which step causing some to be prevented and others not. Still conjecture on my part.

Instead of looking for proof of a negative, we'd just need one instance of something showing it does happen. that shouldn't be to terrible to pull off. grab a sleep pot and fight a chigoe, see if you can parry it. If you can thats a strong indication that all defensive skills (with a skill value) work in like fashion to guard, and we'll need to see whats up with just counter.

**Comment by isotor** [comment]
That's a fair point on the three having associated skills, willing to concede on those until I can prove otherwise.

As to capturing an instance of procs occurring that doesn't help as proc can already occur on DSP - max (I think) is asking me to submit proof of procs not occurring, and I have no idea how I'm supposed to satisfy that outside of community knowledge such as countering in the MNK guide linked above.

**Comment by maxtherabbit** [comment]
you are the one who opened the issue, so it's not unreasonable to ask you to substantiate the premise of your issue.

you provided evidence that counter is wrong, so we can fix that but counter is an offensive ability - it requires the player to be able to hit the monster

i.e. if guard is working correctly on DSP, what reason do you have to suggest that block and parry are different?

**Comment by isotor** [comment]
Fair enough, updated the issue to specifically target countering for now - if I come across similar for the others I'll open a separate issue.

**Comment by isotor** [comment]
[This old PR](https://github.com/DarkstarProject/darkstar/issues/2731) suggests block shouldn't be possible under terror, but parry is fine under sleep.
